src/zope/app/apidoc/codemodule/browser/introspector.rst

```diff
@@ -151,7 +151,7 @@ Now the introspector will show the file and allow you to click on it:
               </b>
               <br />
               <a href="++items++file.txt/@@introspector.html">
-                <code>&lt;zope.app.file.file.File object at ...&gt;</code>
+                <code>&lt;...File object at ...&gt;</code>
```

Comment:
(You know, I kind of miss the fully-qualified dotted names in reprs.)

Reply:
Me too. I would love to consistently get them back, but I don't know the right way to do that in C. Any ideas would be appreciated. https://github.com/zopefoundation/persistent/issues/92

Reply:
How did that work before the introduction of `Persistent.__repr__`?  Did `object.__repr__` report fully-qualified type names?  Isnt't that implemented in C?  Can we steal the implementation?

Reply:
`object.__repr__` was handling it, yes. The issue is that it's somewhat complex and varies between Python versions and apparently even the kind of type object. There's a proposal for a new API in CPython 3.8 that simplifies it, but that obviously doesn't help us. We'd have to steal it.

Reply:
Okay, there's a bit of logic in https://github.com/python/cpython/blob/3.7/Objects/typeobject.c#L3737-L3764 and its helpers.

Could we instead use Python-level API to do the equvalent of `getattr(self, '__module__')` via `PyObject_GetAttrString` and use that instead of checking HEAPTYPE flags etc?

Reply:
possibly? We'd also need to do the same for `__name__` and then do the string concatenation?

Reply:
That seems to work...https://github.com/zopefoundation/persistent/pull/97